### PR TITLE
update oe-selftests to use wait-until-provisioned feature

### PR DESCRIPTION
**Pull request by Zee314159**
Just pushed up the changes. Still testing.

### CODE REVIEW

**Comment by Zee314159** [comment]
When I try to set up environment and bitbake, this happened:

liping@shovel:~/meta-updater-repo/_master$ source meta-updater/scripts/envsetup.sh qemux86-64
-bash: /home/liping/meta-updater-repo/updater_repo_master/meta-updater/scripts/../../poky/oe-init-build-env: No such file or directory
-bash: conf/bblayers.conf: No such file or directory
-bash: conf/bblayers.conf: No such file or directory
-bash: conf/bblayers.conf: No such file or directory
-bash: conf/local.conf: No such file or directory

**Comment by patrickvacek** [comment]
@Zee314159 Hard to say without seeing your directory structure, but it looks like your layers aren't where we expect them to be. Let's move to chat.

File: lib/oeqa/selftest/cases/testutils.py [hunk 1]
```diff
@@ -124,12 +124,18 @@ def verifyProvisioned(testInst, machine):
...
         stdout, stderr, retcode = testInst.qemu_command('aktualizr-info')
-        if retcode == 0 and stderr == b'' and stdout.decode().find('Fetched metadata: yes') >= 0:
+        if retcode == 0 and stderr == b'':
             ran_ok = True
             break
         sleep(delay)
     testInst.assertTrue(ran_ok, 'aktualizr-info failed: ' + stderr.decode() + stdout.decode())
 
+    ran_ok = False
+    if stdout.decode().find('Fetched metadata: yes') == 0:
```
Comment (patrickvacek) [inline lib/oeqa/selftest/cases/testutils.py]: This should be `< 0` to check for metadata not yet found.

Comment (Zee314159) [reply lib/oeqa/selftest/cases/testutils.py]: Ah, thanks!

**Comment by Zee314159** [comment]
The tests passed locally.

CI error message:

$ docker pull "$BITBAKE_IMAGE" || docker pull "$BITBKAE_IMAGE_MASTER" || true
Cannot connect to the Docker daemon at tcp://docker:2375. Is the docker daemon running?
Cannot connect to the Docker daemon at tcp://docker:2375. Is the docker daemon running?
$ docker build --pull --cache-from "$BITBKAE_IMAGE_MASTER" --cache-from "$BITBAKE_IMAGE" -f ./scripts/ci/Dockerfile.bitbake -t "$BITBAKE_IMAGE" ./scripts/ci
time="2019-07-23T13:25:35Z" level=error msg="failed to dial gRPC: cannot connect to the Docker daemon. Is 'docker daemon' running on this host?: dial tcp 172.17.0.2:2375: connect: connection refused"
Cannot connect to the Docker daemon at tcp://docker:2375. Is the docker daemon running?



ERROR: Job failed: exit code 1

**Comment by patrickvacek** [comment]
Sounds like you need to rebase on master due to the problem that Laurent fixed in https://github.com/advancedtelematic/meta-updater/pull/560.

**Pull request closed without merging**